controllers/mysqlcluster_controller.go

```diff
@@ -913,14 +921,43 @@ func (r *MySQLClusterReconciler) reconcileV1StatefulSet(ctx context.Context, req
 		return nil
 	}
 
+	needRecreate := false
+
+	// Recreate StatefulSet if VolumeClaimTemplates has differences.
+	// sts will never be nil.
+	if origApplyConfig != nil && origApplyConfig.Spec != nil && origApplyConfig.Spec.VolumeClaimTemplates != nil {
+		if !equality.Semantic.DeepEqual(sts.Spec.VolumeClaimTemplates, origApplyConfig.Spec.VolumeClaimTemplates) {
+			needRecreate = true
+
+			// Don’t delete the Pod, only delete the StatefulSet.
+			// Same behavior as `kubectl delete sts moco-xxx --cascade=orphan`
+			opt := metav1.DeletePropagationOrphan
+			if err := r.Delete(ctx, &orig, &client.DeleteOptions{
+				PropagationPolicy: &opt,
```

Comment:
envtest is not able to test using `DeletePropagationOrphan` because kube-controller-manager does not exist.
Will verify when adding the E2E test.